### PR TITLE
Bump asm version to 9.6

### PR DESCRIPTION
**Pull request by ov7a**
Complimentary to
* #27868 

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 1]
```diff
@@ -52,6 +52,10 @@ Apache SSHD has been updated from 2.0.0 to https://mina.apache.org/sshd-project/
 
 http://www.jcraft.com/jsch/[JSch] has been replaced by https://github.com/mwiede/jsch[`com.github.mwiede:jsch`] and updated from 0.1.55 to https://github.com/mwiede/jsch/releases/tag/jsch-0.2.16[0.2.16]
 
+==== Upgrade to ASM 9.6
+
+ASM was upgraded from 9.5 to https://asm.ow2.io/versions.html[9.6] for better support of multi-release jars.
```
Comment (asodja) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:57]:
```suggestion
ASM was upgraded from 9.6 to https://asm.ow2.io/versions.html[9.6] for better support of multi-release jars.
```

**Review by asodja — APPROVED** [review]
LGTM

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc (hunk 1 above)
Comment (ov7a) [reply platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:57]: it's "from"?

Comment (asodja) [reply platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:57]: Nevermind, it says `from` and not to. Ignore that

Comment (asodja) [reply platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:57]: My bad :)

Comment (ov7a) [reply platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:57]: NP:)

**Comment by ov7a** [comment]
@bot-gradle merge

**Pull request merged**